libraries/AP_RCProtocol/AP_RCProtocol_DSM.cpp

```diff
@@ -474,10 +474,10 @@ void AP_RCProtocol_DSM::_process_byte(uint32_t timestamp_ms, uint8_t b)
 }
 
 // support byte input
-void AP_RCProtocol_DSM::process_byte(uint8_t b, uint32_t baudrate)
+void AP_RCProtocol_DSM::process_byte(uint8_t b, uint32_t baudrate, uint32_t timestamp_us)
 {
     if (baudrate != 115200) {
         return;
     }
-    _process_byte(AP_HAL::millis(), b);
+    _process_byte(timestamp_us, b);
```

Comment:
this doesn't look right, you can't replace a ms value with a us value. We should change DSM to use micros instead of millis internally

Reply:
The DSM code internally divides by 1000 for the pulse decoding, so we could do the same here.

Reply:
mhmm I overlooked this one, I'll divide it by 1000